src/widget/wsearchlineedit.cpp

```diff
@@ -214,7 +237,7 @@ void WSearchLineEdit::resizeEvent(QResizeEvent* e) {
     }
     int top = rect().top() + m_frameWidth;
     if (layoutDirection() == Qt::LeftToRight) {
-        m_clearButton->move(rect().right() - m_innerHeight - m_frameWidth, top);
+        m_clearButton->move(rect().right() - m_innerHeight - m_frameWidth - m_dropButtonWidth, top);
```

Comment:
in a right-to-left layout the Down button would stay on the right side?
if not, the move() below needs to be adapted, too.
or just freeze the layout direction in the setup: `->setLayoutDirection(Qt::LeftToRight);` then we can get rid of the layout adpation hack which doesn't work with some Clear icons anyway..

Reply:
It behaves like before, just moves the button left of the arrow.

Reply:
I will try test a RightToLeft layout today.

Reply:
I added the offset, but could not test it.

Reply:
Testiing with `setLayoutDirection(Qt::RightToLeft);` is not sufficient, a native RTL locale seem to be necessary.
For the Clear button to be positioned correctly we would also need to update updateClearButton() and clearButtonStyleSheet() to take the layout direction into account.
Right now I wonder if the hazzle is worth it since the skins don't react to the leyout direction..